File: compiler/haskell-ide-core/src/Development/IDE/Compat.hs
-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
-- SPDX-License-Identifier: Apache-2.0

{-# LANGUAGE CPP #-}

-- | Attempt at hiding the GHC version differences we can.
module Development.IDE.Compat(
    HieFile(..),
    mkHieFile,
    writeHieFile,
    readHieFile
    ) where

#ifndef GHC_STABLE
import HieBin
import HieAst
import HieTypes
#else

import GHC
import GhcPlugins
import NameCache
import Avail
import TcRnTypes


mkHieFile :: ModSummary -> TcGblEnv -> RenamedSource -> Hsc HieFile
mkHieFile = undefined

writeHieFile :: FilePath -> HieFile -> IO ()
writeHieFile = undefined

readHieFile :: NameCache -> FilePath -> IO (HieFile, ())
readHieFile _ _ = return (HieFile () [], ())

data HieFile = HieFile {hie_module :: (), hie_exports :: [AvailInfo]}

#endif

File: ledger/participant-state/kvutils/src/test/lib/2.12/com/daml/ledger/participant/state/kvutils/committer/Compat.scala
// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.participant.state.kvutils.committer

private[committer] object Compat {
  def wrapMap[K, V](m: Map[K, V]): Map[K, Option[V]] = new Map[K, Option[V]] {
    override def +[V1 >: Option[V]](kv: (K, V1)): Map[K, V1] = ???
    override def get(key: K): Option[Option[V]] = Some(m.get(key))
    override def iterator: Iterator[(K, Option[V])] = ???
    override def -(key: K): Map[K, Option[V]] = ???
  }
}
